parsers/regex/regex_test.go

```diff
@@ -136,25 +159,52 @@ var tlms = []testLineMap{
 	},
 	{
 		// Sample nginx error log line
-		lineRegex: `(?P<time>\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}) \[(?P<status>.*)\].* request: "(?P<request>[^"]*)"`,
-		input:     `2017/11/07 22:59:46 [error] 5812#0: *777536449 connect() failed (111: Connection refused) while connecting to upstream, client: 127.0.0.1, server: localhost, request: "GET /isbns HTTP/1.1", upstream: "http://127.0.0.1:8080/isbns", host: "localhost"`,
+		lineRegexes: []string{
+			`(?P<time>\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}) \[(?P<status>.*)\].* request: "(?P<request>[^"]*)"`,
+		},
+		input: `2017/11/07 22:59:46 [error] 5812#0: *777536449 connect() failed (111: Connection refused) while connecting to upstream, client: 127.0.0.1, server: localhost, request: "GET /isbns HTTP/1.1", upstream: "http://127.0.0.1:8080/isbns", host: "localhost"`,
 		expected: map[string]interface{}{
 			"time":    "2017/11/07 22:59:46",
 			"status":  "error",
 			"request": "GET /isbns HTTP/1.1",
 		},
 	},
+	{
+		// Multi-regex parsing
+		lineRegexes: []string{
+			`(?P<word1>\w+) (?P<word2>\w+) (?P<word3>\w+)`,
+			`(?P<dummy>banana)`,
+		},
+		input: `apple banana orange`, // Should match to first regex
+		expected: map[string]interface{}{
+			"word1": "apple",
+			"word2": "banana",
+			"word3": "orange",
+		},
+	},
+	{
+		// Multi-regex parsing
+		lineRegexes: []string{
+			`\[(?P<word1>\w+)\]`,
+			`(?P<dummy>banana)`,
+		},
+		input: `apple banana orange`, // Should match to second regex
+		expected: map[string]interface{}{
+			"dummy": "banana",
+		},
+	},
 }
 
 func TestParseLine(t *testing.T) {
-	p := &Parser{
-		conf: Options{
-			NumParsers: 5,
-		},
-		lineParser: &RegexLineParser{regexp.MustCompile(``)},
-	}
 	for _, tlm := range tlms {
-		p.lineParser = &RegexLineParser{regexp.MustCompile(tlm.lineRegex)}
+		p := &Parser{}
+		err := p.Init(&Options{
+			NumParsers: 5,
+			LineRegex:  tlm.lineRegexes,
+		})
+		if err != nil {
```

Comment:
totally optional but `leash.go` uses the `stretchr/testify/assert` library, which has `assert.NoError(...)` for exactly this kind of test - that the function did not return an error. https://godoc.org/github.com/stretchr/testify/assert#NoError